### ISSUE TITLE
Stroked circles are not closed?

### ISSUE THREAD

**Issue by icefoxen** [ggez/ggez#161]
See `examples/graphics_settings.rs` in commit 0c5ee550f59f6cea60260810eee1796a7b1742fd

Probably a lyon bug; verify and punt upstream.

**Comment by icefoxen** [comment]
`graphics::circle(ctx, DrawMode::Line(3.0), Point2::new(400.0, 300.0), 100.0, 4.0)?;` results in this:

![image](https://user-images.githubusercontent.com/1335133/30786671-b30d0d38-a147-11e7-82cc-83474c89d2e7.png)

`graphics::circle()` is:

```rust
pub fn circle(ctx: &mut Context,
              mode: DrawMode,
              point: Point2,
              radius: f32,
              tolerance: f32)
              -> GameResult<()> {
    let m = Mesh::new_circle(ctx, mode, point, radius, tolerance)?;
    m.draw(ctx, Point2::origin(), 0.0)
}
```

`Mesh::new_circle()` is:

```rust
    pub fn new_circle(ctx: &mut Context,
                      mode: DrawMode,
                      point: Point2,
                      radius: f32,
                      tolerance: f32)
                      -> GameResult<Mesh> {
        let mut mb = MeshBuilder::new();
        mb.circle(mode, point, radius, tolerance);
        mb.build(ctx)
    }
```

`MeshBuilder::circle()` is:

```rust
    pub fn circle(&mut self,
                  mode: DrawMode,
                  point: Point2,
                  radius: f32,
                  tolerance: f32)
                  -> &mut Self {
        {
            let buffers = &mut self.buffer;
            match mode {
                DrawMode::Fill => {
                    // These builders have to be in separate match arms 'cause they're actually
                    // different types; one is GeometryBuilder<StrokeVertex> and the other is
                    // GeometryBuilder<FillVertex>
                    let builder = &mut t::BuffersBuilder::new(buffers, VertexBuilder);
                    t::basic_shapes::fill_circle(t::math::point(point.x, point.y),
                                                 radius,
                                                 tolerance,
                                                 builder);
                }
                DrawMode::Line(line_width) => {
                    let builder = &mut t::BuffersBuilder::new(buffers, VertexBuilder);
                    let options = t::StrokeOptions::default()
                        .with_line_width(line_width)
                        .with_tolerance(tolerance);
                    t::basic_shapes::stroke_circle(t::math::point(point.x, point.y),
                                                   radius,
                                                   &options,
                                                   builder);
                }
            };
        }
        self
    }
```

**Comment by icefoxen** [comment]
Ellipse and polyline appears to work.  Circle works when it is drawn in Fill mode instead of Line.

![image](https://user-images.githubusercontent.com/1335133/30786776-fb5cacb4-a148-11e7-97bf-53b177db2e3a.png)

**Comment by icefoxen** [comment]
Fixed in upstream, will be released in lyon 0.8

**Comment by icefoxen** [comment]
Fixed in upstream \o/ updated in commit d4e9451ab41e47cc4f8e551afd34eae1ac375af9
